aptos-move/aptos-vm/src/sharded_block_executor/local_executor_shard.rs

```diff
@@ -0,0 +1,249 @@
+                let (messages_tx, messages_rx) = unbounded();
+                current_shard_msg_txs.push(messages_tx);
+                current_shard_msg_rxs.push(messages_rx);
+            }
+            cross_shard_msg_txs.push(current_shard_msg_txs);
+            cross_shard_msg_rxs.push(current_shard_msg_rxs);
+        }
+        let executor_shards = command_rxs
+            .into_iter()
+            .zip(result_txs.into_iter())
+            .zip(cross_shard_msg_rxs.into_iter())
+            .enumerate()
+            .map(|(shard_id, ((command_rx, result_tx), cross_shard_rxs))| {
+                Self::new(
+                    shard_id as ShardId,
+                    num_shards,
+                    num_threads,
+                    command_rx,
+                    result_tx,
+                    cross_shard_msg_txs.clone(),
+                    cross_shard_rxs,
+                )
+            })
+            .collect();
+        LocalExecutorClient::new(command_txs, result_rxs, executor_shards)
+    }
+}
+
+pub struct LocalExecutorClient<S: StateView + Sync + Send + 'static> {
+    // Channels to send execute block commands to the executor shards.
+    command_txs: Vec<Sender<ExecutorShardCommand<S>>>,
+    // Channels to receive execution results from the executor shards.
+    result_rxs: Vec<Receiver<Result<Vec<Vec<TransactionOutput>>, VMStatus>>>,
+
+    executor_services: Vec<LocalExecutorService<S>>,
+}
+
+impl<S: StateView + Sync + Send + 'static> LocalExecutorClient<S> {
+    pub fn new(
+        command_tx: Vec<Sender<ExecutorShardCommand<S>>>,
+        result_rx: Vec<Receiver<Result<Vec<Vec<TransactionOutput>>, VMStatus>>>,
+        executor_shards: Vec<LocalExecutorService<S>>,
+    ) -> Self {
+        Self {
+            command_txs: command_tx,
+            result_rxs: result_rx,
+            executor_services: executor_shards,
+        }
+    }
+}
+
+impl<S: StateView + Sync + Send + 'static> ExecutorClient<S> for LocalExecutorClient<S> {
+    fn num_shards(&self) -> usize {
+        self.command_txs.len()
+    }
+
+    fn execute_block(
+        &self,
+        state_view: Arc<S>,
+        block: Vec<SubBlocksForShard<AnalyzedTransaction>>,
+        concurrency_level_per_shard: usize,
+        maybe_block_gas_limit: Option<u64>,
+    ) {
+        for (i, sub_blocks_for_shard) in block.into_iter().enumerate() {
+            self.command_txs[i]
+                .send(ExecutorShardCommand::ExecuteSubBlocks(
+                    state_view.clone(),
+                    sub_blocks_for_shard,
+                    concurrency_level_per_shard,
+                    maybe_block_gas_limit,
+                ))
+                .unwrap();
+        }
+    }
+
+    fn get_execution_result(&self) -> Result<Vec<Vec<Vec<TransactionOutput>>>, VMStatus> {
+        trace!("LocalExecutorClient Waiting for results");
+        let mut results = vec![];
+        for rx in self.result_rxs.iter() {
+            results.push(rx.recv().unwrap()?);
```

Comment:
expect("Did not receive output from shard {}", i), where we can get i by enumerating above could be more helpful if smt goes wrong?